### PR TITLE
Pass context to Thrift post response callback

### PR DESCRIPTION
**Pull request by robskillington**
cc @prashantv

### CODE REVIEW

File: thrift/thrift_test.go [hunk 1]
```diff
@@ -250,8 +257,9 @@ func TestRegisterPostResponseCB(t *testing.T) {
 		}
 
 		called := make(chan struct{})
-		cb := func(method string, response thrift.TStruct) {
+		cb := func(reqCtx context.Context, method string, response thrift.TStruct) {
 			assert.Equal(t, "Call", method)
+			assert.Equal(t, ctx.Value(key), value)
```
Comment (prashantv) [inline thrift/thrift_test.go]: nit: `Equal` should have expected argument before the value it's testing, so this should be,
```go
assert.Equal(t, value, ctx.Value(key))
```

File: thrift/thrift_test.go [hunk 2]
```diff
@@ -237,7 +237,14 @@ func TestClientHostPort(t *testing.T) {
 }
 
 func TestRegisterPostResponseCB(t *testing.T) {
-	withSetup(t, func(ctx Context, args testArgs) {
+	key := "key"
```
Comment (prashantv) [inline thrift/thrift_test.go]: nit: can we call these `ctxKey` and `ctxValue` so it's clearer in the rest of the test

File: thrift/thrift_test.go (hunk 1 above)
Comment (prashantv) [inline thrift/thrift_test.go]: I don't think this test is testing what you're intending for it to test. So right now, you're using the `contextFn` to create the context that's passed in to the to the test function. So you're basically testing that your test logic creates the context using the function you specified, and that it has the value you set.

What you actually want to test is: if you call `SetContextFn` (https://godoc.org/github.com/uber/tchannel-go/thrift#Server.SetContextFn) to override the context creation to include an extra value, the context you receive inside of the `cb` has that same value.

You probably want your `contextFn` to be used to call `SetContextFn` on a `thrift.Server` right after creating it.

**Comment by prashantv** [comment]
The library change looks good, but the test isn't testing the right behaviour.

File: thrift/thrift_test.go (hunk 1 above)
Comment (robskillington) [inline thrift/thrift_test.go]: Yeah I just wanted to test its the same context being threaded through, don't really mind about where exactly it's created from.

File: thrift/thrift_test.go (hunk 2 above)
Comment (robskillington) [inline thrift/thrift_test.go]: Sure thing.

File: thrift/thrift_test.go (hunk 1 above)
Comment (prashantv) [inline thrift/thrift_test.go]: The issue is that `reqCtx` is not being tested -- you could pass `nil` as a parameter to the context in the Thrift code, and the test would still pass.

The test should make sure that `reqCtx` is the same context that is returned from the context creation function set by `SetContextFn`

**Comment by robskillington** [comment]
Updated @prashantv with changes, was a lot easier to test with `SetContextFn` that I originally had believed.

**Comment by prashantv** [comment]
lgtm, thanks @robskillington

**Pull request merged**